app/src/main/java/com/example/pinkorange/vibrato/LiveWithSettings.java

```diff
@@ -188,31 +182,12 @@ private void loudnessEnhance() {
         }
     }
 
-    private void requestVisualizerPermissions() {
```

Comment:
Just checking, is this code deletion residual from the previous patch to do with the specific visualizer we're using? Or is it something to do with the added live music functionality?

Reply:
moved requestPermissions to mainactivity to request at start of the app - so it doesn't halt and ask for permissions when you load music

Reply:
Hmm I think the permissions request should happen right before we absolutely need to access those permissions from a user experience perspective. This is because if we ask the user for permissions right at the start of the app, before they know what value the app may provide for them, they will probably be more likely to refuse the permissions than if we ask them for those permissions when we actually need to use them. They might be be more likely to grant those permissions when they are in the flow of performing an action that actually requires granting the app that access.
Secondly, as a future item, I think we should try to create a custom permissions request with some context to educate the user about why exactly we are requesting these permissions and what they are used for, to build trust with the user.
What do you think?

Reply:
sounds good!